src/pages/ProfilePage/index.tsx

```diff
@@ -0,0 +1,190 @@
+import { useContext, useState, useEffect } from "react";
+import { AuthContext } from "../../AuthContext";
+import { Button } from "../../components/Button/index";
+import {
+  addDoc,
+  collection,
+  deleteDoc,
+  doc,
+  getDocs,
+  updateDoc,
+} from "firebase/firestore";
+import { db } from "../../firebase";
+import { QuoteCard } from "../../components/QuoteCard";
+import { useQuotesContext } from "../../QuotesContextProvider";
+
+export const ProfilePage = () => {
+  const authContext = useContext(AuthContext);
+  const [quoteInput, setQuoteInput] = useState<string>("");
+  const [authorInput, setAuthorInput] = useState<string>("");
+  const [message, setMessage] = useState<string | null>(null);
+
+  const [editQuoteId, setEditQuoteId] = useState<string | null>(null);
+  const [editQuoteInput, setEditQuoteInput] = useState("");
+  const [editAuthorInput, setEditAuthorInput] = useState("");
+  const [addNewQuote, setaddNewQuote] = useState<boolean>(false);
+
+  const quotes = useQuotesContext();
+  const userEmail = authContext?.user?.email?.toUpperCase().split("@")[0];
+  const userID = authContext?.user?.uid;
+  const isMessagePositive = message?.includes("unsuccessfully") == false;
+
+  useEffect(() => {
+    if (message) {
+      const timeout = setTimeout(() => {
+        setMessage(null);
+      }, 3000);
+
+      return () => clearTimeout(timeout);
+    }
+  }, [message]);
+
+  async function addQuote(e: React.FormEvent) {
+    try {
+      e.preventDefault();
+      await addDoc(collection(db, "quotes"), {
+        author: authorInput,
+        quote: quoteInput,
+        likedBy: 0,
+        whoAddQuote: userID,
+      });
+
+      setMessage("Quote added successfully!");
+      setQuoteInput("");
+      setAuthorInput("");
+    } catch (error) {
+      setMessage("Quote added unsuccessfully!");
+      console.log(error + "Quote added unsuccessfully!");
+    }
+  }
+
+  async function deleteQuote(id: string) {
+    try {
+      await deleteDoc(doc(db, "quotes", id));
+      setMessage("Quote deleted successfully!");
+    } catch (error) {
+      setMessage("Quote deleted unsuccessfully!!");
+      console.error("Quote deleted unsuccessfully! ", error);
+    }
+  }
+
+  async function updateQuote(id: string, newAuthor: string, newQuote: string) {
+    const quoteRef = doc(db, "quotes", id);
```

Comment:
Before updating the quote, it would be good to check that the user that added the quote initially is the same use that tries to update the quote. You can do that by comparing whoAddQuote property on Quote (better would be to name it 'createdBy') with the id of the currently logged in user.